src/parsers/perf/perfparser.cpp

```diff
@@ -1182,7 +1190,8 @@ class PerfParserPrivate : public QObject
     Data::EventResults eventResult;
     QHash<qint32, QHash<qint32, QString>> commands;
     QScopedPointer<QTextStream> perfScriptOutput;
-    QSet<qint32> reportedMissingDebugInfoModules;
+    QHash<qint32, qint32> numTotalSymbolsByModule;
```

Comment:
please introduce a

```
struct SymbolCount {
    qint32 total = 0;
    qint32 missing = 0;
};
```
and then use that in one single hash, instead of having two hashes

Reply:
I actually thought about doing that. The minor drawback is that you have to iterate over all modules in the end, filtering to those where `missing > 0`. In contrast, right now I can just iterate over `numMissingSymbolsByModule` and only get modules with missing symbols. That's why I kept it the way it is now.

Of course, that efficiency gain is surely negligible with the expected number of modules in a data set. So if you still prefer your suggested version I can change the code to use it. Should I?

Reply:
yes please change it as proposed. in the iteration loop, just do something like:

```
if (!counter.missing)
    continue;
```

Reply:
Sure, will do